lib/active_model/serializer/attributes.rb

```diff
@@ -0,0 +1,59 @@
+module ActiveModel
+  class Serializer
+    # Defines the attributes to be serialized.
+    #
+    module Attributes
+      extend ActiveSupport::Concern
+
+      included do |base|
+        base.class_attribute :_attributes
+        base._attributes ||= []
+        base.class_attribute :_attributes_keys
+        base._attributes_keys ||= {}
+      end
+
+      module ClassMethods
+        def inherit_attributes(base)
+          base._attributes = _attributes.dup
+          base._attributes_keys = _attributes_keys.dup
+        end
+
+        def attributes(*attrs)
+          attrs = attrs.first if attrs.first.class == Array
+
+          attrs.each do |attr|
+            attribute(attr)
+          end
+        end
+
+        def attribute(attr, options = {}, &block)
+          key = options.fetch(:key, attr)
+          _attributes_keys[attr] = { key: key } if key != attr
+          _attributes << key unless _attributes.include?(key)
+
+          ActiveModelSerializers.silence_warnings do
+            define_method key do
```

Comment:
1) we should be conservative in adding new interfaces because then we need to support it
2) if we need this feaure, why? If we don't need it, what are the trade offs of adding it? Not adding it?
3) how does it affect ease of understand usage? Ease of maintenaince? Complexity of either?
4) is it extending existing behavior or modifying it? Could it bd done another way?
5) how do you see this feature evolving?  What are its limitations?  Can it be easily customized? Eg adapter no longer need to be in thd ams namespace or even inherit the base.  So, it's easy to extend (though the boundaries could be clearer and the interfaces more consitent.  Am::s should probably be called a resource and implement the serializers::json interface
(Writing in phone; apologies for anything unclear)
>